app/content/blog/building-beyond-beginner-guitar.md

```diff
@@ -0,0 +1,248 @@
+
+## One-click login
+
+This would add actual security by combining guest comets with email for a one-click login.
+
+For a bit of technical background, when someone visits an Urbit served site, `%eyre` sets a cookie and presents the browser session as a comet. But this _guest_ comet has your `@p` tacked onto the end. ex: `~tillus-nodhus-nidsud-pontug--rammeb-sicpur-nordus-mocwyl`
+
+In effect, it's a fingerprint for that browser that lasts seven days. As long as they don't open up another browser that person, as far as your Urbit is concerned will be that comet for seven days.
+
+The process could look something like this:
++ Purchase the course with Stripe.
++ Use the provided email.
++ Upon opening the course request the email.
++ Send a transactional email with an email service.
++ Open email with one-click login link.
++ Clicking the link maps the email to the relevant guest comet.
++ This will allow access from this browser session for 7 days.
++ After 7 days they'll need another one-click login email.
+
+This requires using an email service that can handle transactional emails, which is a bit of a bummer. But not only would this allow security in my access control, it would also open the door for personalization and progress tracking.
+
+## Service-specific micro hosting provider
+
+Another option I can pursue today attacks the "If everybody was on Urbit, we can make `this` thing, awesome!" problem by, well, onboarding more people on to their own instance of Urbit OS. I could have a dozen of ships running and ready to go. Prebooted, hawk installed, guitar course imported, group membership accepted, and roles assigned.
+
+We could still do the same Stripe payment, or integrate crypto payments more deeply. But the product they are buying is actually a hosted ship. I would essentially act as a small host provider for a specific service, the guitar course. Urbit being a computer in itself is incidental, but present and extensible. 
+
+Instead of purchasing access to a secret URL, purchasing the guitar course grants access to a one of these Urbit ships that has the guitar course pre-installed. 
+
+I could do this with a moon, if it seems like users aren't ready for full self-sovereignty, or I could sell them a planet from my star. Perhaps branching between the two based on if they used a Stripe or a crypto payment method. And even with the moon-based users, I could assign them a moon that was connected to the equivalent planet, earmarked from my star's children, e.g.`~sampel-palnet-dozzod-forryx` could be assigned to a credit card user, and when they are ready to upgrade then `~forryx` could issue `~sampel-palnet` to a self-custody wallet. And all this could be coordinated on Urbit via the moon, rather than depending on email.
```

Comment:
I could offer either a moon for users not yet ready for full self-sovereignty, or sell them a planet from my star. I could then branch between the two options based on whether they used Stripe or a crypto payment method. Even for moon-based users, I could assign them a moon connected to an equivalent planet earmarked from my star's children. For example, `~tortus-sopbel-dozzod-forryx` could be assigned to a credit card user; when they are ready to upgrade, `~forryx` could issue `~tortus-sopbel` to their self-custody wallet. Crucially, all coordination and communication for this process could be handled on Urbit via the moon, eliminating the dependency on email.